src/registry.js

```diff
@@ -74,6 +100,65 @@ class inViewRegistry {
         return this;
     }
 
+    /**
+    * Validates the options passed, falls back to defaults
+    */
+    validate(o) {
+        // create local copy of defaults
+        let options = merge({}, defaults);
+        // if parameter is not an object, return
+        if (typeof o !== 'object') return options;
+        // validate options
+        Object.keys(options).forEach(prop => {
+            if (o.hasOwnProperty(prop)) {
+                options[prop] = this[prop].bind({
+                    options: options
+                })(o[prop]);
+            }
+        });
```

Comment:
hard to figure out what's going on here. you're going over the option keys and then calling the corresponding functions and assigning the return value back as the option value?

Reply:
yea, so you can pass `{ offset: 0 }` and would work the same `.offset(0)`, or if you pass an unsupported value i.e `threshold: 2`, it fallbacks to the defaults.

Reply:
The difference is that if you pass different options for the same selector, it doesn't override it but creates another unique selector, so 2 different files can target `.embed-youtube` with 2 different options/handlers. Without it, one of those would be forced to do the lookup first, and pass them as nodes. 

The bad thing about nodes is that if you pass an array, there's no easy way to know the contents of that array already exist on 1 selector without looping through all and do equal checks on the values, otherwise the node counter would bump automatically, and you could have multiple entries with the same elements.

A way to get around it would be to set the registry in a variable i.e `Node-5`, and the elements yourself `registry.elements.concat([..]);`, or add a method that does that for you.